### PR TITLE
chore: Update dependencies and move CI to Github Actions

### PR DESCRIPTION
**Pull request by JKRhb**
This PR updates the project's dependencies to their latest version, adds `nyc` for code coverage determination, and moves the CI from Travis to Github Actions.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 16.x]
```
Comment (mcollina) [inline .github/workflows/test.yml:16]: drop 10.x it is out of maintenance.

File: package.json [hunk 2]
```diff
@@ -27,11 +28,12 @@
...
-    "chai": "~1.8.0",
-    "mocha": "~1.13.0",
-    "pre-commit": "0.0.3",
-    "typescript": "~2.2.2"
+    "@types/node": "~16.9.1",
+    "chai": "~4.3.4",
+    "mocha": "^9.1.1",
+    "nyc": "^15.1.0",
+    "pre-commit": "1.2.2",
+    "typescript": "~4.4.3"
```
Comment (mcollina) [inline package.json:36]: please use ^ everywhere for the dpendencies.

File: .github/workflows/test.yml (hunk 1 above)
Comment (JKRhb) [reply .github/workflows/test.yml]: Both of your comments should now be addressed :)

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**